src/ds/champ_map.h

```diff
@@ -490,13 +456,18 @@ namespace champ
     {
       return root->foreach(0, std::forward<F>(f));
     }
+
+    std::unique_ptr<Snapshot<K, V, H>> make_snapshot() const
+    {
+      return std::make_unique<Snapshot<K, V, H>>(*this);
+    }
```

Comment:
Slightly nicer interface to get a snapshot on a map (also available for the `RBMap`).